services/galley/src/Galley/API/MLS/Message.hs

```diff
@@ -360,9 +360,11 @@ processCommit qusr con lconv epoch sender commit = do
         -- client (the creator)
         case (sender, first (toList . lmMLSClients) self) of
           (MemberSender ref, Left [creatorClient]) -> do
+            -- use update path as sender reference and if not existing fall back to sender
+            let senderRef = maybe ref (fromMaybe ref . kpRef' . upLeaf) $ cPath commit
```

Comment:
Cannot this be collapsed into one `maybe` instead of a combination of `fromMaybe` and `maybe`? The fact you're using `ref` in two default cases suggests it is possible.

Reply:
Disregard this comment; I see you changed it in a subsequent commit.